backend/src/data_entry/mod.rs

```diff
@@ -1,38 +1,95 @@
+use crate::data_entry::repository::PollingStationDataEntries;
+use crate::election::repository::Elections;
+use crate::error::{APIError, ErrorReference, ErrorResponse};
+use crate::polling_station::repository::PollingStations;
+use crate::polling_station::structs::PollingStation;
 use axum::extract::{FromRequest, Path, State};
 use axum::http::StatusCode;
 use axum::response::{IntoResponse, Response};
 use axum::Json;
+use chrono::{DateTime, Utc};
 use entry_number::EntryNumber;
 use serde::{Deserialize, Serialize};
-use sqlx::SqlitePool;
+use status::{ClientState, DataEntryStatus, DataEntryStatusName};
 use utoipa::ToSchema;
 
-use crate::data_entry::repository::{PollingStationDataEntries, PollingStationResultsEntries};
-use crate::election::repository::Elections;
-use crate::error::{APIError, ErrorReference, ErrorResponse};
-use crate::polling_station::repository::PollingStations;
-use crate::polling_station::structs::PollingStation;
-
 pub use self::structs::*;
 pub use self::validation::*;
 
-mod entry_number;
+pub mod entry_number;
 pub mod repository;
-mod structs;
+pub mod status;
+pub mod structs;
 mod validation;
 
+/// Response structure for getting data entry of polling station results
+#[derive(Serialize, Deserialize, ToSchema, Debug)]
+pub struct GetDataEntryResponse {
+    pub progress: u8,
+    pub data: PollingStationResults,
+    #[schema(value_type = Object)]
+    pub client_state: Option<serde_json::Value>,
+    pub validation_results: ValidationResults,
+    #[schema(value_type = String)]
+    pub updated_at: DateTime<Utc>,
+}
+
+/// Get an in-progress (not finalised) data entry for a polling station
+#[utoipa::path(
+    get,
+    path = "/api/polling_stations/{polling_station_id}/data_entries/{entry_number}",
+    responses(
+        (status = 200, description = "Data entry retrieved successfully", body = GetDataEntryResponse),
+        (status = 404, description = "Not found", body = ErrorResponse),
+        (status = 500, description = "Internal server error", body = ErrorResponse),
+    ),
+    params(
+        ("polling_station_id" = u32, description = "Polling station database id"),
+        ("entry_number" = u8, description = "Data entry number (first or second data entry)"),
+    ),
+)]
+pub async fn polling_station_data_entry_get(
+    State(polling_station_data_entries): State<PollingStationDataEntries>,
+    State(polling_stations): State<PollingStations>,
+    State(elections): State<Elections>,
+    Path((id, _entry_number)): Path<(u32, EntryNumber)>, // note: we don't need the entry number here
```

Comment:
You can make the comment a `TODO` and mention #762.

Reply:
Fixed in https://github.com/kiesraad/abacus/pull/750/commits/123c18020c6c3f22faf9cac76bc43e1941e7bc0a